pkg/cmd/cli/cmd/set/imagelookup.go

```diff
@@ -0,0 +1,235 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/cmd/templates"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+var (
+	imageLookupLong = templates.LongDesc(`
+		Use an image stream from pods and other objects
+
+		Image streams make it easy to tag images, track changes from other registries, and centralize
+		access control to images. Local name lookup allows an image stream to be the source of
+		images for pods, deployments, replica sets, and other resources that reference images, without
+		having to provide the full registry URL. If	local name lookup is enabled for an image stream
+		named 'mysql', a pod or other resource that references 'mysql:latest' (or any other tag) will
+		pull from the location specified by the image stream tag, not from an upstream registry.
+
+		Once lookup is enabled, simply reference the image stream tag in the image field of the object.
+		For example:
+
+		    $ oc import-image mysql:latest --confirm
+				$ %[1]s image-lookup mysql
+				$ oc run mysql --image=mysql
+
+		will import the latest MySQL image from the DockerHub, set that image stream to handle the
+		"mysql" name within the project, and then launch a deployment that points to the image we
+		imported.`)
+
+	imageLookupExample = templates.Examples(`
+		# Print all of the image streams and whether they resolve local names
+	  %[1]s image-lookup
+
+	  # Use local name lookup on image stream mysql
+	  %[1]s image-lookup mysql
+
+	  # Disable local name lookup on image stream mysql
+	  %[1]s image-lookup mysql --enabled=false
+
+	  # Set local name lookup on all image streams
+	  %[1]s image-lookup --all`)
+)
+
+type ImageLookupOptions struct {
+	Out io.Writer
+	Err io.Writer
+
+	Filenames []string
+	Selector  string
+	All       bool
+
+	Builder *resource.Builder
+	Infos   []*resource.Info
+
+	Encoder runtime.Encoder
+
+	ShortOutput   bool
+	Mapper        meta.RESTMapper
+	OutputVersion unversioned.GroupVersion
+
+	PrintTable  bool
+	PrintObject func(runtime.Object) error
+
+	Enabled bool
+}
+
+// NewCmdImageLookup implements the set image-lookup command
+func NewCmdImageLookup(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ImageLookupOptions{
+		Out:     out,
+		Err:     errOut,
+		Enabled: true,
+	}
+	cmd := &cobra.Command{
+		Use:     "image-lookup STREAMNAME [...]",
```

Comment:
I think of this in terms of "what am I doing"

1. oc set triggers (set my triggers)
2. oc set route-backends (set the backends for a route)
3. oc set image (set the image)

When I put this in my head, `oc set lookup-strategy` doesn't quite match to what this is doing.  We're setting how "image resolution" or "image lookup" is handled.  Resolution is kind of a big word, and we use that for a more specific meaning now ("when you resolve a tag to an image, resolutionPolicy is applied"), so I don't want to overlap between "oc set image-resolution".  

So to a user, they're defining "how image lookup on pods, replica sets, and jobs should be handled".  However, "NameLookup" as the field didn't really seem right - generally we try to have declarative fields, so we say what the flag does ("importPolicy", "insecurePolicy", "from", "to").  I think we are trying to indicate to the person editing the image stream that "this stream" can be used to "handle image references without a repository" or "resolve the name of a ??? image when asked".  We don't really have a great way to describe what an image name like `mysql:foo` is - docker uses a couple of names that don't fit to how we're using to them.